strawberry/annotation.py

```diff
@@ -61,6 +61,9 @@ def __eq__(self, other: object) -> bool:
 
         return self.resolve() == other.resolve()
 
+    def __hash__(self) -> int:
+        return hash(self.resolve())
```

Comment:
can you add a test for this? 😊

Reply:
Sure thing!

Reply:
While making the tests I found a couple more classes were the hashing function was broken. I hope the collateral changes are fine! 😅